src/librustc_mir/interpret/machine.rs

```diff
@@ -232,6 +242,20 @@ pub trait Machine<'mir, 'tcx>: Sized {
         id
     }
 
+    /// In Rust, thread locals are just special statics. Therefore, the compiler
```

Comment:
As Ralf already commented, these work the same in codegen as in miri, so these docs aren't quite right